# Section 1 : Business Understadning
Context :  I am a travel agent , advising first time travellers to seatle. Below are some the questions that my customers often ask me, and I use data science to give them answers

**Question 01 :** Which neighborhood in Seattle should I (Traveller) choose ?

**Question 02 :** Which time of the year should I travel in ?

**Question 03 :** How much should I budget for accomodation ?





In [2]:
import pandas as pd
import numpy as np 
from datetime import datetime
import matplotlib.pyplot as plt 
%matplotlib inline


Gather data

In [3]:
#Read files
listings_df = pd.read_csv(r'C:\Users\rammr001\OneDrive - Vodafone Group\Documents\Data Science\dataset\listings.csv')
#listings_df.head()
calendar_df = pd.read_csv(r'C:\Users\rammr001\OneDrive - Vodafone Group\Documents\Data Science\dataset\calendar.csv')
#calendar_df.head
reviews_df = pd.read_csv(r'C:\Users\rammr001\OneDrive - Vodafone Group\Documents\Data Science\dataset\reviews.csv')
#reviews_df.head()

# Section 02 : Data understanding

Get info on the Data

In [4]:
listings_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3818 entries, 0 to 3817
Data columns (total 92 columns):
id                                  3818 non-null int64
listing_url                         3818 non-null object
scrape_id                           3818 non-null int64
last_scraped                        3818 non-null object
name                                3818 non-null object
summary                             3641 non-null object
space                               3249 non-null object
description                         3818 non-null object
experiences_offered                 3818 non-null object
neighborhood_overview               2786 non-null object
notes                               2212 non-null object
transit                             2884 non-null object
thumbnail_url                       3498 non-null object
medium_url                          3498 non-null object
picture_url                         3818 non-null object
xl_picture_url                      3498

In [5]:
calendar_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1393570 entries, 0 to 1393569
Data columns (total 4 columns):
listing_id    1393570 non-null int64
date          1393570 non-null object
available     1393570 non-null object
price         934542 non-null object
dtypes: int64(1), object(3)
memory usage: 42.5+ MB


In [6]:
reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84849 entries, 0 to 84848
Data columns (total 6 columns):
listing_id       84849 non-null int64
id               84849 non-null int64
date             84849 non-null object
reviewer_id      84849 non-null int64
reviewer_name    84849 non-null object
comments         84831 non-null object
dtypes: int64(3), object(3)
memory usage: 3.9+ MB


Define a funcutionto describe a column

In [7]:
def describe_column (column) :
    ''' returns the description of the data in column
        input : 
            column : a single column from a dataframe

        output : 
            stats : a data frame object showing statistics of the column
    '''

    stats = column.describe()
    
    return stats


# Section 3 Data Preparation

Check if any columns have null values

In [8]:
listings_df.columns[listings_df.isnull().any()]

Index(['summary', 'space', 'neighborhood_overview', 'notes', 'transit',
       'thumbnail_url', 'medium_url', 'xl_picture_url', 'host_name',
       'host_since', 'host_location', 'host_about', 'host_response_time',
       'host_response_rate', 'host_acceptance_rate', 'host_is_superhost',
       'host_thumbnail_url', 'host_picture_url', 'host_neighbourhood',
       'host_listings_count', 'host_total_listings_count',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'zipcode', 'property_type', 'bathrooms', 'bedrooms', 'beds',
       'square_feet', 'weekly_price', 'monthly_price', 'security_deposit',
       'cleaning_fee', 'first_review', 'last_review', 'review_scores_rating',
       'review_scores_accuracy', 'review_scores_cleanliness',
       'review_scores_checkin', 'review_scores_communication',
       'review_scores_location', 'review_scores_value', 'license',
       'reviews_per_month'],
      dtype='object')

Missing values

Since I am not making any predicitions , missing values dont affect my analysis as much

Conver date column into a date time object to perform nifty operations

In [9]:
#convert date column from type object to type datetime
calendar_df["date"] = pd.to_datetime(calendar_df["date"])

In [10]:
#Check if the conversion worked
calendar_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1393570 entries, 0 to 1393569
Data columns (total 4 columns):
listing_id    1393570 non-null int64
date          1393570 non-null datetime64[ns]
available     1393570 non-null object
price         934542 non-null object
dtypes: datetime64[ns](1), int64(1), object(2)
memory usage: 42.5+ MB


# Section 4 Evaluate results

Question 01 : Which neighbourhood should I choose ?

In [11]:
# Use the describe function to describe the neighbourhood column
neighbourhood_df =describe_column(listings_df[['neighbourhood']])
neighbourhood_df

neighbourhood
count           3402
unique            81
top     Capitol Hill
freq             351

We can see the top neighborhood is Capitial hill, this means that Capitol hill is the neighbourhood with the most listings, providing you with the most options for a better experiece

**Question 02:** Which time of the year should I travel ?

In [12]:
#Create a simpler way to view just the month from the data column, by creating an additonal coloumn called month
calendar_df['month']=calendar_df['date'].dt.month
calendar_df.head()

listing_id       date available   price  month
0      241032 2016-01-04         t  $85.00      1
1      241032 2016-01-05         t  $85.00      1
2      241032 2016-01-06         f     NaN      1
3      241032 2016-01-07         f     NaN      1
4      241032 2016-01-08         f     NaN      1

In [13]:
#Create a subst dataframe to view available months
x = calendar_df[calendar_df['available']=='t']
x[['available','month']]

available  month
0               t      1
1               t      1
9               t      1
10              t      1
14              t      1
...           ...    ...
1393207         t      1
1393208         t      1
1393211         t      1
1393212         t      1
1393213         t      1

[934542 rows x 2 columns]

In [14]:
#Which month of the year 
x[['month']].mode()


month
0     12

As we can see the month with the highest availibily is 12, That means the best time of the year to travel is in December.

**Question 03:** How much should I budget

In [15]:
#Find the maxmimum price in data
calendar_df = calendar_df.dropna()
Budget = describe_column(calendar_df[['price']])
Budget

price
count    934542
unique      669
top     $150.00
freq      36646

As we can see that the top prices is $150, so a good budget for accomdation in seatle is $150 per night